packages/worker/src/rxutils.ts

```diff
@@ -27,6 +27,19 @@ export function mergeMapWithState<T, I, O>(
   );
 }
 
+export function mapWithState<T, I, O>(
+  fn: (state: T, input: I) => StateAndOutput<T, O>,
+  initialState: T
+): OperatorFunction<I, O> {
```

Comment:
:eyes: Is that you, `fold`?

Reply:
It's very much like `scan` (uses scan internally)